src/atomate2/abinit/powerups.py

```diff
@@ -224,3 +228,94 @@ class filter will match any subclasses.
     }
 
     return update_maker_kwargs(class_filter, dict_mod_updates, flow, name_filter)
+
+
+def update_taskdoc(
+    flow: Job | Flow | Maker,
+    taskdoc_updates: dict[str, Any],
+    name_filter: str | None = None,
+    class_filter: type[Maker] | None = BaseAbinitMaker,
+) -> Job | Flow | Maker:
+    """
+    Update any attribute of any AbinitTaskDoc in the flow.
+
+    Alternatively, if a Maker is supplied, the attributes of the maker will
+    be updated.
+
+    Note, this returns a copy of the original Job/Flow/Maker. I.e., the update does not
+    happen in place.
+
+    Parameters
+    ----------
+    flow : .Job or .Flow or .Maker
+        A job, flow or Maker.
+    taskdoc_updates : dict
+        The updates to apply to the input generator.
+    name_filter : str or None
+        A filter for the name of the jobs.
+    class_filter : Maker or None
+        A filter for the BaseAbinitMaker class used to generate the flows. Note the
+        class filter will match any subclasses.
+
+    Returns
+    -------
+    Job or Flow or Maker
+        A copy of the input flow/job/maker modified to use the updated factory settings.
+    """
+    dict_mod_updates = {
+        f"task_document_kwargs->{k}": v for k, v in taskdoc_updates.items()
+    }
+
+    return update_maker_kwargs(class_filter, dict_mod_updates, flow, name_filter)
+
+
+def update_clean_flow(
+    flow: Job | Flow,
+    exclude_files_from_zip: list[str | Path] | None = None,
+    delete: bool = True,
+    exclude_files_from_del: list[str | Path] | None = None,
+    include_files_to_del: list[str | Path] | None = None,
+) -> Job | Flow | Maker:
+    r"""
+    Append a job to delete and/or compress files of the flow.
+
+    Note, this returns a copy of the original Job/Flow/Maker. I.e., the update does not
+    happen in place.
+
+    Parameters
+    ----------
+    flow : .Job or .Flow
+        A job, or flow.
+    exclude_files_from_zip
+        Filenames to exclude from the compression.
+        Supports glob file matching, e.g., "\*.dat".
+    delete: bool = True,
+        Activates the deletion of the files or not.
+    exclude_files_from_del
+        Filenames to exclude from the compression.
+        Supports glob file matching, e.g., "\*.dat".
+    include_files_to_del
+        Filenames to include as a list of str or Path objects given relative to
+        directory. Glob file paths are supported, e.g. "\*.dat". If ``None``, all files
+        in the directory will be deleted.
+
+    Returns
+    -------
+    Job or Flow
+        A copy of the input flow/job/maker modified to delete/compress files
+        once completed.
+    """
+    copied_flow = deepcopy(flow)
+    return Flow(
+        [
+            copied_flow,
+            del_gzip_files(
+                copied_flow.output,
```

Comment:
Yes I thought that it would make sense to force the flows to have a determined output, namely the list of output of its jobs. Why don't we want this in the long term?
Otherwise I can implement your suggestion by iterating over the flow/job, it is a good idea

Reply:
let's iterate over the flow to avoid initiating a new convention of output (not uniformized for all codes)

Reply:
done